pkg/common/cns-lib/vsphere/virtualmachine.go

```diff
@@ -337,3 +337,38 @@ func (vm *VirtualMachine) IsInZoneRegion(ctx context.Context, zoneCategoryName s
 func GetUUIDFromProviderID(providerID string) string {
 	return strings.TrimPrefix(providerID, providerPrefix)
 }
+
+// GetSharedDatastoresForVMs returns shared datastores accessible to specified
```

Comment:
No changes were made to this function. It has just been copied from `nodes.go` to `virtualmachine.go`